service/api/service_impl.go

```diff
@@ -149,6 +151,111 @@ func (s serviceImpl) PublishToLocalQueue(
 	return nil
 }
 
+func (s serviceImpl) Rpc(
+	ctx context.Context, request xdbapi.ProcessExecutionRpcRequest,
+) (response *xdbapi.ProcessExecutionRpcResponse, retErr *ErrorWithStatus) {
+	latestPrcExe, err := s.store.GetLatestProcessExecution(ctx, data_models.GetLatestProcessExecutionRequest{
+		Namespace: request.GetNamespace(),
+		ProcessId: request.GetProcessId(),
+	})
+	if err != nil {
+		return nil, s.handleUnknownError(err)
+	}
+
+	if latestPrcExe.NotExists || latestPrcExe.Status != data_models.ProcessExecutionStatusRunning {
```

Comment:
in IWF, a read-only RPC is allowed for a closed workflow execution. Do you think we can still support it?

Reply:
Sounds good. I moved the second checking condition later to skip writing if the process is not running after receiving the RPC response.